### DISCUSSION TITLE
Pro/Con of Postgres Data Source Relationships vs Relationship field type in Budibase?

### DISCUSSION THREAD

**Discussion by pricedev** [Budibase/budibase#6629] [Help]
Very new to Budibase and in the process of setting up my Postgres database for a project.  I was wondering what were the pros / cons of defining relationships via the "Relationships" section of the Postgres data source panel vs using relationship column types from within in Budibase.

I'm assuming the Postgres data source route creates the definition in the actual postgres database, while the relationship type in budibase results in budibase maintaining the definition within itself.  Assuming that is correct, is there any benefit of one over the other with respect to developing the application?

**Comment by melohagan** [comment]
Hi @pricedev 

This is a great question! 

I'll start off by stating that I strongly recommend that you use the *Relationships* section of the Postgres data source panel. 

This is because the schema should already have been built and designed outside of Budibase if you are using an external data source.
The relationships section is there to allow Budibase to effectively wrap the existing design that has already been established.
This will create consistency between the Budibase use of Postgres, and any other applications/platforms that could be using the same Postgres DB.

With that said, there may be some scenarios where you want to include a relationship between your Postgres tables *only* within Budibase, and without affecting the Postgres schema as used by other applications/platforms.
This wont take advantage of SQL queries however, and should be used sparingly.
